marconi-chain-index/app/Main.hs

```diff
@@ -1,51 +1,33 @@
-{-# LANGUAGE OverloadedStrings   #-}
-{-# LANGUAGE ScopedTypeVariables #-}
-
+{-# LANGUAGE OverloadedStrings #-}
 module Main where
 
-import Control.Exception (catch)
-import Prettyprinter (defaultLayoutOptions, layoutPretty, pretty, (<+>))
-import Prettyprinter.Render.Text (renderStrict)
-
 import Cardano.Api qualified as C
-import Cardano.BM.Setup (withTrace)
-import Cardano.BM.Trace (logError)
-import Cardano.BM.Tracing (defaultConfigStdout)
-import Cardano.Streaming (ChainSyncEventException (NoIntersectionFound), withChainSyncEventStream)
 import Marconi.ChainIndex.CLI qualified as Cli
-import Marconi.ChainIndex.Indexers (filterIndexers, mkIndexerStream, startIndexers)
-import Marconi.ChainIndex.Logging (logging)
+import Marconi.ChainIndex.Indexers (filterIndexers, mkIndexerStream, runIndexers, startIndexers)
 import System.Directory (createDirectoryIfMissing)
 
 main :: IO ()
 main = do
   o <- Cli.parseOptions
   createDirectoryIfMissing True (Cli.optionsDbPath o)
-  c <- defaultConfigStdout
-  withTrace c "marconi" $ \trace -> do
-    let indexers = filterIndexers (Cli.utxoDbPath o)
-                                  (Cli.addressDatumDbPath o)
-                                  (Cli.datumDbPath o)
-                                  (Cli.scriptTxDbPath o)
-                                  (Cli.epochStakepoolSizeDbPath o)
-                                  (Cli.mintBurnDbPath o)
-                                  (Cli.optionsTargetAddresses o)
-                                  (Cli.optionsNodeConfigPath o)
-    (cp, coordinator) <- startIndexers indexers
-    let preferredChainPoints =
-          -- If the user specifies the chain point then use that,
-          -- otherwise use what the indexers provide.
-          if Cli.optionsChainPoint o == C.ChainPointAtGenesis
-             then cp
-             else [Cli.optionsChainPoint o]
-    withChainSyncEventStream
-      (Cli.optionsSocketPath o)
-      (Cli.optionsNetworkId o)
-      preferredChainPoints
-      (mkIndexerStream coordinator . logging trace)
-      `catch` \NoIntersectionFound ->
-        logError trace $
-          renderStrict $
-            layoutPretty defaultLayoutOptions $
-              "No intersection found when looking for the chain point" <+> (pretty . Cli.optionsChainPoint) o <> "."
-                <+> "Please check the slot number and the block hash do belong to the chain"
+  let indexers = filterIndexers (Cli.utxoDbPath o)
+                                (Cli.addressDatumDbPath o)
+                                (Cli.datumDbPath o)
+                                (Cli.scriptTxDbPath o)
+                                (Cli.epochStakepoolSizeDbPath o)
+                                (Cli.mintBurnDbPath o)
+                                (Cli.optionsTargetAddresses o)
+                                (Cli.optionsNodeConfigPath o)
+  (returnedCp, coordinator) <- startIndexers indexers
```

Comment:
Rename `startIndexers` to `initialiseIndexers`?

Also, seems like we can bundle all code lines from 13 into a the library portion of `marconi-chain-index`?

Reply:
a42e0cfe8